### PR TITLE
91 subdomain

### LINKED ISSUE #91: [Wish] Add custom subdomains for each board
Companies are more likely to share their boards if the domain includes their name. 

www.wyshlist.net/organizations/10
is not as good as
lewagon.wyshlist.com

I don't think we need to isolate the database with a multi-tenant app. Just by doing a subdomain redirect should be enough, but not 100% sure in this case. To discuss @christinasugiono

### PR DESCRIPTION
**Pull request by christinasugiono**
close #91 

- Subdomain is implemented after user sign up/sign in
- Subdomain is using the organization name (All wishlist belongs to an organization)
- For development: the url needs to changed to `lvh.me` (instead of `localhost:3000`)

Screenshot 2023-07-21 at 14 53 29

### CODE REVIEW

**Review by Danrod16 — CHANGES_REQUESTED** [review]
This is amazing @christinasugiono ! You went super fast? So will we have different databases for each organization as we had with Avenist? Or is this a different setup? Also could we write a couple of tests to make sure that after creating and organization or joining we are indeed redirected to a subdomain? 

Thanks!

**Comment by christinasugiono** [comment]
> This is amazing @christinasugiono ! You went super fast? So will we have different databases for each organization as we had with Avenist? Or is this a different setup? Also could we write a couple of tests to make sure that after creating and organization or joining we are indeed redirected to a subdomain?
> 
> Thanks!

I didn't separate the database, just constraining the routes to have subdomain, and check the subdomain to be the company name
Please help check and play around with it first 😄

File: app/models/organization.rb [hunk 1]
```diff
@@ -17,4 +17,26 @@ def organization_owner?(user)
...
+          # find_by(subdomain: request.params[:subdomain])
+        rescue StandardError
+          nil
+        end
+      else
+        # all.map(&:subdomain).include? request.subdomain
+        nil
+        # find_by(subdomain: request.subdomain)
+      end
+      raise
```
Comment (AGorinam) [inline app/models/organization.rb]: 👀

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -17,6 +18,12 @@ def user_not_authorized
     redirect_to(authenticated_root_path)
   end
 
+  def check_subdomain
+    if user_signed_in? && (request.subdomain != current_user.organization.subdomain)
+      redirect_to authenticated_root_url(subdomain: current_user.organization.subdomain), allow_other_host: true
+    end
```
Comment (AGorinam) [inline app/controllers/application_controller.rb:24]: might be wrong but what if I am a visitor and want to check one random company's wishlists, will I be redirected?

Comment (christinasugiono) [reply app/controllers/application_controller.rb:24]: I haven't take that into account! Thanks for pointing it out @AGorinam !
So the condition is they can visit other company's wishlist that are public?

Comment (Danrod16) [reply app/controllers/application_controller.rb:24]: It feels like we should create helpers for these conditionals, we are starting to have many, but we will do that in a different pull request :)

**Comment by Danrod16** [comment]
Bug when visiting organization show page. I created the user with avatar and organization with logo.

**Comment by Danrod16** [comment]
@christinasugiono The screenshot was the typical error we all had that the user needed an organization 👍

**Pull request merged**